Source/WebCore/bindings/js/JSDOMAsyncIterator.h

```diff
@@ -111,6 +108,21 @@ template<typename JSWrapper, typename IteratorTraits, typename JSIterator> class
     {
     }
 
+    JSC::EncodedJSValue settle(JSC::JSGlobalObject*);
+    JSC::JSBoundFunction* createOnSettledFunction(JSC::JSGlobalObject*);
+    static JSC::EncodedJSValue onPromiseSettledBody(JSC::JSGlobalObject* globalObject, JSC::CallFrame*, JSDOMAsyncIteratorBase* castedThis) { return castedThis->settle(globalObject); }
+    static JSC::EncodedJSValue onPromiseSettled(JSC::JSGlobalObject*, JSC::CallFrame*);
```

Comment:
This does not work. All JS functions need to be defined via JSC_DECLARE_HOST_FUNCTION etc.. See JSDOMIterator::next's JSC_HOST_CALL_ATTRIBUTES and CodeGenerator.pm's JSC_ANNOTATE_HOST_FUNCTION for each next function.